openedx/frontend-component-header
# feat: display profile name in navbar if available

Display profile name if it is set, else fallback to username.

* `Private-ref`: [BB-8772](https://tasks.opencraft.com/browse/BB-8772)

**Test instructions:**

* Setup tutor-quince locally.
* Create a plugin in your `tutor plugins printroot` directory called `mfe_custom_header`.
* Add below content to the file (We have published the changes in npm with a new package)
```python
from tutor import hooks

hooks.Filters.ENV_PATCHES.add_item(
    (
        "mfe-dockerfile-post-npm-install",
        """
RUN --mount=type=cache,target=/root/.npm,sharing=shared echo "installing custom header" && npm install @edx/frontend-component-header@npm:@opencraft/frontend-component-header@~4.6.1 && echo "done"
""",
    )
)
```
* Enable this plugin by running `tutor plugins enable mfe_custom_header`.
* Save tutor config by running `tutor config save`.
* Run `tutor images build mfe`.
* And `tutor dev launch -I`.

Now the new header should be installed in all MFEs (except dashboard as it doesn't depend on header component). Login with any user and if the user profile name is not set in `Account` page, you will see `username` like before. Update profile name by in the account setting page and logout and login again to see the name appear in header right corner.
